### PR TITLE
DT-1493 and DT-1494 - List/Describe Batch Operations

### PR DESCRIPTION
**Pull request by rossedfort**
## Description & motivation 💭 
Add pages to list batch operations and view an individual batch operation, as well as a link to the list page in the nav. Also update the UX when creating a batch operation, instead of differentiation between "batch" and "bulk", we simply show a toast that says the request is processing in the background, and show an animation in the batch icon in the side nav
### Screenshots (if applicable) 📸 
localhost_3000_namespaces_canary_batch-operations_ad3ea083-76e7-466f-b55f-95162248b70c (1)
localhost_3000_namespaces_canary_batch-operations_ad3ea083-76e7-466f-b55f-95162248b70c

https://github.com/temporalio/ui/assets/11775628/6efb4f98-e9e5-4c9f-9a42-9873ccf2d54d

### How was this tested 👻 

- [x] Manual testing
- [ ] E2E tests added
- [ ] Unit tests added

### CODE REVIEW

**Comment by rossedfort** [comment]
Not planning on implementing pagination until https://github.com/temporalio/temporal/issues/4907 is addressed

File: src/routes/(app)/namespaces/[namespace]/batch-operations/[jobId]/+layout.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+import type { LayoutLoad } from '../$types';
+
+export const load: LayoutLoad = async ({ params, fetch }): Promise<void> => {
+  const operation = await describeBatchOperation(
+    { namespace: params.namespace, jobId: params.jobId },
+    fetch,
+  );
+
+  batchOperation.set(operation);
```
Comment (Alex-Tideman) [inline src/routes/(app)/namespaces/[namespace]/batch-operations/[jobId]/+layout.ts:12]: Why not just return the operation and use $page.data.operation?

File: src/routes/(app)/namespaces/[namespace]/batch-operations/+layout.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { listBatchOperations } from '$lib/services/batch-service';
+import { batchOperations } from '$lib/stores/batch-operations';
+
+import type { LayoutLoad } from '../$types';
+
+export const load: LayoutLoad = async ({ params, fetch }): Promise<void> => {
+  const { operations } = await listBatchOperations(params.namespace, fetch);
+
```
Comment (Alex-Tideman) [inline src/routes/(app)/namespaces/[namespace]/batch-operations/+layout.ts:8]: Same, why not return operations and have $page.data.operations

File: src/lib/services/batch-service.ts [hunk 3]
```diff
@@ -186,30 +168,86 @@ export async function pollBatchOperation({
...
+      10,
+    ),
+    failureOperationCount: parseInt(
+      apiBatchOperationDetails.failureOperationCount,
+      10,
+    ),
+  };
+};
+
+export async function listBatchOperations(
```
Comment (Alex-Tideman) [inline src/lib/services/batch-service.ts:224]: Just for my own education, do these ever go away after a certain period (retention period)? Or this call will return all batch operations of all time?

File: src/routes/(app)/namespaces/[namespace]/batch-operations/[jobId]/+layout.ts (hunk 1 above)
Comment (Alex-Tideman) [reply src/routes/(app)/namespaces/[namespace]/batch-operations/[jobId]/+layout.ts:12]: Also, we'll need to remember to add these +layout files to cloud. which is totally fine

Comment (rossedfort) [reply src/routes/(app)/namespaces/[namespace]/batch-operations/[jobId]/+layout.ts:12]: Yeah true, we could do that. This does make auto-refresh easier cause then I can just update this store value. I'll tinker with it a little.

File: src/lib/layouts/batch-operation.svelte [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+{#if $inProgressBatchOperation}
+  <Alert
+    icon="info"
+    intent="info"
+    title={translate('batch', 'max-concurrent-alert-title')}
+  >
+    {translate('batch', 'max-concurrent-alert-description')}
+  </Alert>
+{/if}
```
[the remark this answers is not in the document]
Comment (rossedfort) [reply src/lib/layouts/batch-operation.svelte:15]: Fair point, it is displayed on both the list and details page though. I'll make it better.

File: src/lib/pages/batch-operation.svelte [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  import { autoRefresh, batchOperation } from '$lib/stores/batch-operations';
+  import { routeForBatchOperations } from '$lib/utilities/route-for';
+
+  export let namespace: string;
+
+  let interval: number;
+
+  $: {
+    if (
+      $autoRefresh === 'on' &&
```
[the remark this answers is not in the document]
Comment (rossedfort) [reply src/lib/pages/batch-operation.svelte:20]: Agree, but this is a persisted store, so I don't think we can store booleans, or if we did, we'd have to stringify/parse them each time we read the store value.

File: src/lib/services/batch-service.ts (hunk 3 above)
Comment (rossedfort) [reply src/lib/services/batch-service.ts:224]: That's a good question, I assume they do not adhere to the namespace's retention period, but I can double check on that.

File: src/lib/layouts/batch-operation.svelte (hunk 4 above)
[the remark this answers is not in the document]
Comment (rossedfort) [reply src/lib/layouts/batch-operation.svelte]: Actually we were only showing it on the list page, so I nuked the layout 👍

File: src/routes/(app)/namespaces/[namespace]/batch-operations/[jobId]/+layout.ts (hunk 1 above)
Comment (rossedfort) [reply src/routes/(app)/namespaces/[namespace]/batch-operations/[jobId]/+layout.ts]: Moved to using page.data and it works fine 👍

File: src/lib/components/side-nav.svelte [hunk 6]
```diff
@@ -64,6 +92,14 @@
         <div class="nav-title">{translate('archive')}</div>
       </NavRow>
     </IsCloudGuard>
```
Comment (Alex-Tideman) [inline src/lib/components/side-nav.svelte:94]: I'd put this NavRow in <IsCloudGuard> until we are ready for it in cloud. Or add a slot="batch"

File: src/lib/pages/batch-operation.svelte (hunk 5 above)
[the remark this answers is not in the document]
Comment (rossedfort) [reply src/lib/pages/batch-operation.svelte:22]: Nvm, made it a boolean

**Review by Alex-Tideman — APPROVED** [review]

**Pull request merged**